app/models/note.rb

```diff
@@ -4,8 +4,17 @@ class Note < ApplicationRecord
   validates :body, presence: true
   belongs_to :submission
 
-  belongs_to :author,
-             foreign_key: :gravity_user_id,
-             class_name: 'User',
-             primary_key: :gravity_user_id
+  def author
+    @author ||= load_author
```

Comment:
I like this! So we have a lazy-loaded, memo-ized author, right? 😎

Reply:
yeah, except it doesn't work because a missing `@author` will always be `nil`, so we have to check with `defined?(@author)` which is much more verbose - cc @yuki24